BLSQ/iaso
# IA-2345: Permissions grouping for IASO admin

Explain what problem this PR is resolving
- Group permissions into modules
- Limit the access of permissions bye  modules

Related JIRA tickets : [IA-2345](Permissions grouping for IASO admin)

## Self proofreading checklist


## Changes
- Added models (Module and ModulePermission(extends Auth/Permission model))
- Link Modules to ModulePermission throught ModulePermission
- Create Modules https://docs.google.com/spreadsheets/d/1zzCD19HJLm3xkDfS3HrZonMvbBPsY9YEXhywDCpoFzU/edit#gid=7786233 throught migrations
- Link Account with modules
- Add modules JsonField on setup_account
- Edit filter_permissions to take in account modules
- Display on the front end modules(those activated for the account and the other not activated

## How to test
- Run migrations(to create the modules and their related permissions)
- Go in admin/iaso/account/
- Link an account to modules
- Go to fronte and try to edit permissions linked to a user or userrole
- Go again the admin menu
- Go to modules menu 
- Check which modules are active and not for the account

## Print screen / video

![Screenshot from 2023-09-14 15-41-39](https://github.com/BLSQ/iaso/assets/19631540/9748c665-9ab6-4b56-b840-8c7dc9a03bc3)

![Screenshot from 2023-09-14 15-43-01](https://github.com/BLSQ/iaso/assets/19631540/3af33746-0112-4e24-bdf6-1627d8333dc1)

## Notes

Things that the reviewers should know: known bugs that are out of the scope of the PR, other trade-offs that were made.
If the PR depends on a PR in bluesquare-components, or merges into another PR (i.o. main), it should also be mentioned here


[IA-2345]: https://bluesquare.atlassian.net/browse/IA-2345?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
